src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+import java.io.File;
+import java.io.IOException;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import java.time.DateTimeException;
+
+import duke.command.Command;
+
+import duke.dukeexception.DukeException;
+
+import duke.parser.Parser;
+
+import duke.storage.Storage;
+
+import duke.task.TaskList;
+
+import duke.ui.Ui;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke() throws IOException {
+        this("data/duke.txt");
+    }
+
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            File file = new File("data");
+            if (!file.exists()) {
+                //create new data dir and duke.file
+                new File("data").mkdir();
+            }
+            new File(filePath).createNewFile();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
```

Comment:
Overall really good abstraction and encapsulation! Only a minor feedback for javadocs, perhaps you could be more consistent with the format of the comments and spacings between methods. Using gradle's checkstyle might help. But overall really clean code!